### ISSUE TITLE
Validator\File\MimeType open_basedir error not handled

### ISSUE THREAD

**Issue by smartexcan** [laminas/laminas-validator#202]
### Bug Report

|    Q        |   A
|------------ | ------
| Version(s)  | 2.35.0

#### Summary

open_basedir errors are not handled in `getMagicFile()`, as it is trying to catch an exception, but no exception is thrown if `is_file()` is called on a file outside the open_basedir restriction.

#### Current behavior

If getMagicFile() tries to auto-discover the magic file from one of the common locations, and one of those files is not in the open_basedir restriction, an error is raised.
https://github.com/laminas/laminas-validator/blob/5fafe1ec4cc23e4bb4dfe6b4cd96c28e1c7f48a3/src/File/MimeType.php#L166-L172
No error handler is set when `is_file()` is called in setMagicFile(), so no exception gets thrown, resulting in the error propagating upward.
```
Warning: is_file(): open_basedir restriction in effect. File(/usr/share/misc/magic) is not within the allowed path(s): (/home/test/:/tmp:/var/tmp:/opt/alt/php81/usr/share/pear/:/dev/urandom:/usr/local/lib/php/:/usr/local/php81/lib/php/) in /home/test/public_html/vendor/laminas/laminas-validator/src/File/MimeType.php on line 208
```

#### How to reproduce

```php
// open_basedir set as "/home/test/:/tmp:/var/tmp:/opt/alt/php81/usr/share/pear/:/dev/urandom:/usr/local/lib/php/:/usr/local/php81/lib/php/"
$mimeValidator = new \Laminas\Validator\File\MimeType([]);
$valid = $mimeValidator->isValid('/home/test/public_html/tmp/test.jpg');
```

#### Expected behavior

open_basedir error should be handled, possibly by using `\Laminas\Stdlib\ErrorHandler` or by other means, and exception should be caught, to allow continuing as intended.

**Comment by Ocramius** [comment]
I don't think this code should be aware of the existence of `open_basedir()` at all: it's an extension that messes with filesystem built-in procedures, and IMO it should not be accounted for when the built-in procedures are integrated as per their original documentation.

**Comment by smartexcan** [comment]
I suppose that is true, and it's not really my decision is there should be code to account for it in here or not.
Only submitting this issue as the code that was already implemented to handle the error didn't actually handle the error.

**Comment by gsteel** [comment]
Magic file options/usage is being removed in 3.0 - ref #340 and this issue won't be resolved in the 2.x series
